### PR TITLE
336 Bugfix/hide add contact when archived

### PR DESCRIPTION
**Pull request by retex73**
This ticket covers preventing certain actions being performed against an archived company 

Pages affected: 
- Subsidiaries
- Contacts 
- Interactions
- Investment
- Orders (OMIS)

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (yeahfro) [inline .vscode/launch.json]: Should this be here?

File: .vscode/launch.json [hunk 2]
```diff
@@ -1,31 +0,0 @@
-{
```
Comment (yeahfro) [inline .vscode/launch.json]: This needs cleaning up with a fixup on a previous commit

File: .gitignore [hunk 3]
```diff
@@ -35,6 +35,7 @@ stats.json
 package-lock.json
 .vscode/launch.json
 .vscode/settings.json
+.vscode/launch.json
```
Comment (yeahfro) [inline .gitignore]: twice?

File: src/apps/companies/views/contacts.njk [hunk 4]
```diff
@@ -1,23 +1,40 @@
...
+  </summary>
+  <div class="details__content" id="details-content-0" aria-hidden="false">
+    <p class="c-message c-message--muted">Contacts cannot be added to an archived company.
+    <a href="/companies/{{company.id}}/unarchive">Click here to unarchive</a>
+        </p>
+  </div>
+</details>
+{% endif %}
+
+{{ CollectionFilters({
```
Comment (yeahfro) [inline src/apps/companies/views/contacts.njk]: Indentation does not look right here

File: src/apps/companies/views/interactions.njk [hunk 5]
```diff
@@ -1,36 +1,51 @@
...
+  </summary>
+  <div class="details__content" id="details-content-0" aria-hidden="false">
+    <p class="c-message c-message--muted">Interactions cannot be added to an archived company.
+      <a href="/companies/{{company.id}}/unarchive">Click here to unarchive</a>
     </p>
+  </div>
+</details>
+{% endif %}
+
+{% block xhr_content %}
```
Comment (yeahfro) [inline src/apps/companies/views/interactions.njk]: Indentation does not look right here

File: test/acceptance/features/step_definitions/common.js [hunk 6]
```diff
@@ -69,6 +69,13 @@ Then(/^I should not see the "([^"]*)?" (link|button)$/, async function (linkText
     .useCss()
 })
 
+Then(/^I should see the "([^"]*)?" (link|button)$/, async function (linkText, type) {
```
Comment (yeahfro) [inline test/acceptance/features/step_definitions/common.js]: This seems like it should be in a different commit?

File: .vscode/launch.json (hunk 1 above)
Comment (retex73) [reply .vscode/launch.json]: I've since removed this file from git. It should never have been added in the first place

Comment (yeahfro) [reply .vscode/launch.json]: We normally clean up the commits with `fixup`. It will be better to apply a `fixup` commit here rather than cleaning up with a different commit.

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (yeahfro) [inline .vscode/settings.json]: Again, better to clean this up with `fixup`

File: .gitignore (hunk 3 above)
Comment (yeahfro) [reply .gitignore]: This needs cleaning up with `fixup`

File: .gitignore [hunk 8]
```diff
@@ -35,7 +35,6 @@ stats.json
 package-lock.json
 .vscode/launch.json
 .vscode/settings.json
-.vscode/launch.json
```
Comment (yeahfro) [inline .gitignore]: If this is cleaned up with `fixup` then it will not be required here.

Comment (yeahfro) [reply .gitignore]: Once the commit has been tidied up then the message can be more specific
